pytensor/link/numba/dispatch/slinalg.py

```diff
@@ -126,17 +126,17 @@ def impl(A, B, trans, lower, unit_diagonal, b_ndim, overwrite_b):
 
         B_is_1d = B.ndim == 1
 
-        if overwrite_b:
-            B_copy = B
-        else:
-            if B_is_1d:
-                # _copy_to_fortran_order does nothing with vectors
-                B_copy = np.copy(B)
-            else:
-                B_copy = _copy_to_fortran_order(B)
+        A_copy = _copy_to_fortran_order(A)
 
-        if B_is_1d:
-            B_copy = np.expand_dims(B_copy, -1)
+        # This list is exhaustive, but numba freaks out if we include a final else clause
```

Comment:
maybe if you nest the conditions: 
```python
if overwrite_b:
  if B_is_1d:
    ...
  else:
    ...
else:
  if ...
```

Not a biggie, just wondering

Reply:
I do find it a bit more readable if you change the second case of the if and statement faster (C-contiguous boolean logic) instead of the F-contiguous you have here:
```python
if a and b
elif a and not b
elif not a and b
elif not a and not b
```
Also a NIT and debatable

Reply:
I like avoiding nesting as a general rule, but I think it went a little too far here. It's much more readable nested.